### ISSUE TITLE
Soft Assert mode with a FAIL step is not marking the whole test as FAILED

### ISSUE THREAD

**Issue by jensenbrian** [selenide/selenide#2755]
## The problem
In Soft Assert mode with a FAIL step is not marking the whole test as FAILED

## Details
Running the `TestNgSoftTest.textArea()` in SOFT mode with an added line that will fail. The build is marked as "BUILD SUCCESSFUL". (see branch testng-softassert-not-failing [HERE](https://github.com/jensenbrian/selenide/blob/dea03bece6e15dab419ee784d122fdbc8711ef34/modules/testng/src/test/java/integration/TestNgSoftTest.java#L64))
* **I think the test should continue as it does but mark the build as failed. Am I missing a basic concept or is this a bug?**

Running the same change made to `TestNgSampleTest.textArea()` in STRICT mode. The test does stop processing and marks the build as "BUILD FAILED". (This is not an issue. Working as expected.)

## Tell us about your environment
* **Selenide Version**: latest = 7.3.2
* **Chrome\Firefox\IE Version**: Chrome Version=125.0.6422.113 Platform=mac
* **Browser Driver Version**: 
* **Selenide-Appium Version**:
* **IOS\Android\WEB version**:
* **Selenium Version**: 4.21.0 build revision: 79ed462ef4
* **OS Version**: macOS Sonoma 14.5

## Code To Reproduce Issue
[modules/testng/src/test/java/integration/TestNgSoftTest.java](https://github.com/jensenbrian/selenide/blob/1dd8d005c217210b5aa080a80aa7766f45b7de6b/modules/testng/src/test/java/integration/TestNgSoftTest.java)

```java
  @Override
  @BeforeSuite
  final void setupAsserts() {
    Configuration.assertionMode = SOFT;
    SoftAsserts.fullStacktraces = false;
    closeWebDriver();
    WebDriverRunner.addListener(listener1);
  }

  @Test
  public void textAreaShouldFail() {
    $("#empty-text-area").val("text for textarea");
    // Will fail (on purpose) test will continue but should be marked as a FAILED test is in SOFT mode
    $("#empty-text-area").shouldNotHave(value("text for textarea"));
    $("#empty-text-area").shouldHave(value("text for textarea"));
  }
```

**Comment by jensenbrian** [comment]
I found in the changelog:
6.6.0 (released 08.06.2022)
#1834 fix SoftAsserts with TestNG (downgraded TestNG from 7.5 to 7.4.0) -- see PR #1843

Looks like with the 7.0.0 release - bump TestNG from 7.4.0 to 7.8.0 (#2515)

I recently updated from 6.19.1 to 7.3.2
Could this still be broken in TestNG? If so should TestNG be downgraded back to 7.4.0?

I will see if I can figure out how to try that.
Thank you for your thoughts/review

**Comment by jensenbrian** [comment]
Thanks @BorisOsipov . Looks like the "right" fix mentioned in #3046 is to use TestNG 7.10.0 and "Ordering listeners in TestNG" documented in [Specifying listeners with ServiceLoader](https://testng.org/#_specifying_listeners_with_serviceloader)

I will see what I can do. "I am working on it"

**Comment by asolntsev** [comment]
@jensenbrian Do we still need to fix anything in Selenide, or can we close the issue?

**Comment by BorisOsipov** [comment]
@asolntsev we can't close for sure. The test like
```
        <dependency>
            <groupId>com.codeborne</groupId>
            <artifactId>selenide-testng</artifactId>
            <version>7.3.3</version>
            <scope>test</scope>
        </dependency>

@Listeners({SoftAsserts.class})
public class Foo {
    @Test
    public void bar() {
        Configuration.assertionMode = AssertionMode.SOFT;
        open("https://www.selenium.dev/");
        $("#PUPA-LUPA").shouldBe(visible);
    }
}
```
is always `PASSED` but should be `FAILED`.

**Comment by asolntsev** [comment]
@BorisOsipov I cannot reproduce the problem. I've copied your code, and the test fails.

Maybe the reason is that Selenide uses TestNG 7.9.0, but you are trying to use some newer version of TestNG?

**Comment by jensenbrian** [comment]
While this is still an issue I agree this is cannot be fixed in Selenide.
I have linked it to an existing gradle issue: 18828
Thanks for your help and consideration

**Comment by asolntsev** [comment]
@jensenbrian What about using older TestNG 7.9.0?
Soft asserts work nicely with this version.

**Comment by jensenbrian** [comment]
Hi @asolntsev 
When we were on 6.19.0 Selenide was using TestNG 7.4.0 and all was good.

A few months ago we upgraded Selenide versioned to 7.0.0 and required Java 17 it also bumped TestNG 7.9.0. This reintroduced this SoftAsserts issue.

I tried 7.90 and it did not work for me using Gradle. Maybe I missed something and I can try this again.

I have tried forcing back to TestNG 7.4.0 but it has a bug in the RetryAnalyzer class which causes other issues for us. For now we are running all tests in STRICT mode. Not good but at least we do not get false positives.
